library/modules/Gui.cpp

```diff
@@ -934,30 +946,138 @@ df::unit *Gui::getAnyUnit(df::viewscreen *top)
         return NULL;
     }
 
+    if (VIRTUAL_CAST_VAR(screen, df::viewscreen_reportlistst, top))
+        return vector_get(screen->units, screen->cursor);
+
+    if (VIRTUAL_CAST_VAR(screen, df::viewscreen_announcelistst, top))
+    {
+        if (screen->unit) {
+            // in (r)eports -> enter
+            auto *report = vector_get(screen->reports, screen->sel_idx);
+            if (report)
+            {
+                for (df::unit *unit : world->units.all)
+                {
+                    if (unit && screen->report_type >= 0 && screen->report_type < 3
+                        && unit != screen->unit) // find 'other' unit related to this report
+                    {
+                        for (int32_t report_id : unit->reports.log[screen->report_type])
+                        {
+                            if (report_id == report->id)
+                                return unit;
+                        }
+                    }
+                }
+            }
+        } else {
+            // in (a)nnouncements
+            return NULL; // cannot determine unit from reports
+        }
+    }
+
+    if (VIRTUAL_CAST_VAR(screen, df::viewscreen_layer_militaryst, top))
+    {
+        if (screen->page == df::viewscreen_layer_militaryst::T_page::Positions) {
+            auto positions = getLayerList(screen, 1);
+            if (positions && positions->enabled && positions->active)
```

Comment:
I think the lists should always exist, even if they're empty, but can't hurt to check.